src/errors/mod.rs

```diff
@@ -0,0 +1,234 @@
+pub mod io;
+
+extern crate url;
+
+use crate::cloud_provider::Kind;
+use url::Url;
+
+pub struct SimpleError {
+    message: String,
+    message_safe: String,
+}
+
+#[derive(Debug)]
+pub enum LogLevel {
+    Debug,
+    Info,
+    Warning,
+    Error,
+    Critical,
```

Comment:
What the difference between Error and Critical ?

Reply:
I would say Critical will fail the process instantly, but maybe Error should too, so I guess I can remove Critical.